### PR TITLE
adding `rank_eval indices` API tests

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
test for:
GET /_rank_eval
POST /_rank_eval
GET /{index}/_rank_eval
POST /{index}/_rank_eval

### Issues Resolved
[#663 ]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
1.3.x returns `_type` in the docs, schema should be referencing an existing document model somewhere

File: spec/schemas/_core.rank_eval.yaml [hunk 1]
```diff
@@ -168,6 +168,8 @@ components:
           $ref: '_common.yaml#/components/schemas/IndexName'
         _score:
           type: number
+        _type:
```
Comment (dblock) [inline spec/schemas/_core.rank_eval.yaml:171]: All these things should be references, like https://github.com/opensearch-project/opensearch-api-specification/blob/3981d332cc830c1547ece4f3eea9ea3399040126/spec/schemas/_core.get.yaml#L12.

**Comment by Tokesh** [comment]
I think its ready for review, open to any suggestions! :)
We have achieved 80% test coverage! 
We're almost there—100% coverage is within reach!

File: spec/schemas/_core.rank_eval.yaml [hunk 2]
```diff
@@ -134,7 +134,7 @@ components:
           additionalProperties:
             type: object
             additionalProperties:
-              type: object
+              type: number
```
[the remark this answers is not in the document]
Comment (Tokesh) [reply spec/schemas/_core.rank_eval.yaml:137]: I'll try to read it in the future, most likely yes.
If I notice it's not, I'll open a new pull request.

**Pull request merged**